### DISCUSSION TITLE
Optimizing node_modules for Lambda

### DISCUSSION THREAD

**Discussion by maoosi** [prisma/prisma#6396] [Q&A] [answered]
After upgrading the [Prisma-AppSync generator](https://github.com/maoosi/prisma-appsync) from Prisma 2.18.0 to Prisma 2.20.1, I started facing issues with the Lambda bundle size - exceeding the hard limit of 250MB unzipped (only with a naked boilerplate and very minimal code).

Up until now, I had `node_modules/prisma`, `node_modules/@prisma` and `node_modules/.prisma` folders as part of the bundled dependencies. To reduce the bundle size, I ended up deleting the query engines located in `node_modules/prisma`:

```typescript
// extract from the Lambda function bundling process
afterBundling() {
    return [
        'npx prisma generate', 
        'rm -rf node_modules/prisma/query-engine-darwin', 
        'rm -rf node_modules/prisma/query-engine-rhel-openssl-1.0.x'
    ]
}
```

Doing the above helped, though the unzipped package is still heavy and dangerously close to the limit:
- `node_modules/@prisma` ~= 96MB
- `node_modules/prisma` ~= 18MB
- `node_modules/.prisma` ~= 83MB

I’d be keen to get some direction on what would be the best approach to reduce bundling size for Production (AWS Lambda env) and limit the dependencies to the bare minimum.

**Comment by ryands17** [comment, marked as answer]
I have brought it down to around 11 mb zipped by removing these folders. These files can be safely deleted: https://github.com/ryands17/prisma-lambda/blob/main/lib/prisma-lambda-stack.ts#L31-L37

**Reply by maoosi** [reply]
Awesome, thanks for the list! Will try this out asap.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
